core/src/main/java/org/infinispan/encoding/DataConversion.java

```diff
@@ -258,7 +253,8 @@ public boolean equals(Object o) {
       DataConversion that = (DataConversion) o;
       return Objects.equals(encoder, that.encoder) &&
             Objects.equals(wrapper, that.wrapper) &&
-            Objects.equals(transcoder, that.transcoder);
+            Objects.equals(transcoder, that.transcoder) &&
+            isKey == that.isKey;
```

Comment:
Putting ```isKey == that.isKey``` first will make the comparison fail faster :)

Reply:
changed!